src/igvfd/schemas/construct_library.json

```diff
@@ -0,0 +1,368 @@
+            "not": {
+                "required": [
+                    "mpra_library_details"
+                ]
+            },
+            "comment": "Specification of crispr_library_details is mutually exclusive with specification of mpra_library_details."
+        },
+        "mpra_library_details": {
+            "not": {
+                "required": [
+                    "crispr_library_details"
+                ]
+            },
+            "comment": "Specification of mpra_library_details is mutually exclusive with specification of crispr_library_details."
+        },
+        "scope": {
+            "oneOf": [
+                {
+                    "required": [
+                        "targeted_genes"
+                    ],
+                    "properties": {
+                        "scope": {
+                            "enum": [
+                                "genes"
+                            ]
+                        }
+                    }
+                },
+                {
+                    "required": [
+                        "targeted_loci"
+                    ],
+                    "properties": {
+                        "scope": {
+                            "enum": [
+                                "loci"
+                            ]
+                        }
+                    }
+                },
+                {
+                    "not": {
+                        "properties": {
+                            "scope": {
+                                "enum": [
+                                    "genes",
+                                    "loci"
+                                ]
+                            }
+                        }
+                    }
+                }
+            ]
+        },
+        "average_size": {
+            "not": {
+                "required": [
+                    "size_range"
+                ]
+            },
+            "comment": "Specification of average_size is mutually exclusive with specification of size_range."
+        },
+        "size_range": {
+            "not": {
+                "required": [
+                    "average_size"
+                ]
+            },
+            "comment": "Specification of size_range is mutually exclusive with specification of average_size."
+        }
+    },
+    "properties": {
+        "schema_version": {
+            "default": "1"
+        },
+        "accession": {
+            "accessionType": "FS"
+        },
+        "scope": {
+            "title": "Library Construct Scope",
+            "description": "The scope or scale that this construct library is designed to target. If the scope is across gene(s) or loci, these will need to be specified in the targeted_genes or targeted_loci property.",
+            "type": "string",
+            "enum": [
+                "genes",
+                "loci",
+                "genome-wide"
+            ]
+        },
+        "origin": {
```

Comment:
I think there could be multiple origins? like library from histone and TF or TF and DNase. I think should be a plural and a list